src/+soilmoisture/calculateTimeDerivatives.m

```diff
@@ -47,21 +47,23 @@
         RHS(n) = -C9(n) - C7(n) + (C1(n - 1, 2) * h(n - 1) + C1(n, 1) * h(n)) / Delt_t ...
           - (C2(n - 1, 2) / Delt_t + C5(n - 1, 2)) * TT(n - 1) - (C2(n, 1) / Delt_t + C5(n, 1)) * TT(n) ...
           + (C2(n - 1, 2) / Delt_t) * T(n - 1) + (C2(n, 1) / Delt_t) * T(n);
+
     elseif ~ModelSettings.Thmrlefc && ModelSettings.Soilairefc
         RHS(indxBotm) = -C9(indxBotm) - C7(indxBotm) + (C1(indxBotm, 1) * h(indxBotm) + C1(indxBotm, 2) * h(indxBotm + 1)) / Delt_t ...
             - C6(indxBotm, 1) * P_gg(indxBotm) - C6(indxBotm, 2) * P_gg(indxBotm + 1);
-        for i = 2:ModelSettings.NL
+        for i = indxBotm:ModelSettings.NL
             RHS(i) = -C9(i) - C7(i) + (C1(i - 1, 2) * h(i - 1) + C1(i, 1) * h(i) + C1(i, 2) * h(i + 1)) / Delt_t ...
                 - C6(i - 1, 2) * P_gg(i - 1) - C6(i, 1) * P_gg(i) - C6(i, 2) * P_gg(i + 1);
         end
         RHS(n) = -C9(n) - C7(n) + (C1(n - 1, 2) * h(n - 1) + C1(n, 1) * h(n)) / Delt_t ...
             - C6(n - 1, 2) * P_gg(n - 1) - C6(n, 1) * P_gg(n);
+
     elseif ModelSettings.Thmrlefc && ModelSettings.Soilairefc
         RHS(indxBotm) = -C9(indxBotm) - C7(indxBotm) + (C1(indxBotm, 1) * h(indxBotm) + C1(indxBotm, 2) * h(indxBotm + 1)) / Delt_t ...
             - (C2(indxBotm, 1) / Delt_t + C5(indxBotm, 1)) * TT(indxBotm) - (C2(indxBotm, 2) / Delt_t + C5(indxBotm, 2)) * TT(indxBotm + 1) ...
             - C6(indxBotm, 1) * P_gg(indxBotm) - C6(indxBotm, 2) * P_gg(2) ...
             + (C2(indxBotm, 1) / Delt_t) * T(indxBotm) + (C2(indxBotm, 2) / Delt_t) * T(indxBotm + 1);
-        for i = 2:ModelSettings.NL
+        for i = indxBotm:ModelSettings.NL
```

Comment:
Running the model fails due to this change. In this file, if `~GroundwaterSettings.GroundwaterCoupling` , then `indxBotm=1`, the equation` ARG1 = C2(i - 1, 2) / Delt_t;` returns an error because i-1 is 0. Can you fix it?

Reply:
> Running the model fails due to this change. In this file, if `~GroundwaterSettings.GroundwaterCoupling` , then `indxBotm=1`, the equation` ARG1 = C2(i - 1, 2) / Delt_t;` returns an error because i-1 is 0. Can you fix it?

My mistake, I correct it now